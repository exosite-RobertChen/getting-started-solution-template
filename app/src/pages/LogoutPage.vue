<template>
  <v-container class="fill-height">
    <v-row class="whitevar--text ml-0" align="center">
      <v-col class="text-center">
        <h1 class="mb-4" :class="$vuetify.breakpoint.xs ? 'headline' : ''">
          You're now signed out
        </h1>
        <p>
          You can go to the
          <router-link
            class="text-underline"
            :to="{ name: 'LandingPage' }"
            data-qa="LandingPageLink"
            >home page</router-link
          >
          or
          <router-link
            class="text-underline"
            :to="{ name: 'LoginPage' }"
            data-qa="LoginPageLink"
            >sign in</router-link
          >
          again.
        </p>
      </v-col>
    </v-row>
  </v-container>
</template>
