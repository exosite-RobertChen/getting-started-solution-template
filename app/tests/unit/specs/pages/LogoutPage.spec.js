import { shallowComponent } from '@/../tests/unit/common'
import LogoutPage from '@/pages/LogoutPage.vue'

const factory = overrides => shallowComponent(LogoutPage, overrides)

describe('LogoutPage.vue', () => {
  let wrapper = factory()
  test('can be mounted', () => {
    expect(wrapper.contains(LogoutPage)).toBe(true)
  })

  test('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('renders landing page and login page links', () => {
    expect(wrapper.contains('[data-qa=LandingPageLink]')).toBe(true)
    expect(wrapper.contains('[data-qa=LoginPageLink]')).toBe(true)
  })
})
