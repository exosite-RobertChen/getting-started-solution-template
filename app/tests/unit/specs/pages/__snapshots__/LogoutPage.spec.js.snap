// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutPage.vue matches snapshot 1`] = `
<v-container-stub
  class="fill-height"
  tag="div"
>
  <v-row-stub
    align="center"
    class="whitevar--text ml-0"
    tag="div"
  >
    <v-col-stub
      class="text-center"
      tag="div"
    >
      <h1
        class="mb-4"
      >
        
        You're now signed out
      
      </h1>
       
      <p>
        
        You can go to the
        
        <router-link-stub
          class="text-underline"
          data-qa="LandingPageLink"
          event="click"
          tag="a"
          to="[object Object]"
        >
          home page
        </router-link-stub>
        
        or
        
        <router-link-stub
          class="text-underline"
          data-qa="LoginPageLink"
          event="click"
          tag="a"
          to="[object Object]"
        >
          sign in
        </router-link-stub>
        
        again.
      
      </p>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
